pkg/reconciler/shared/tektonconfig/upgrade/upgrade_test.go

```diff
@@ -81,15 +115,30 @@ func TestUpdateAppliedUpgradeVersion(t *testing.T) {
 	_, err = ug.operatorClient.OperatorV1alpha1().TektonConfigs().Create(ctx, tc, metav1.CreateOptions{})
 	assert.NoError(t, err)
 
-	// tektonConfig CR present
-	// should return no error
-	err = ug.updateAppliedUpgradeVersion(ctx)
+	// pre upgrade tests
+	// ---
+
+	// tektonConfig CR present, should return no error
+	err = ug.updateAppliedUpgradeVersion(ctx, isPreUpgrade)
 	assert.NoError(t, err)
 
-	// verify the version in tektonConfig CR
+	// verify the pre upgrade version in tektonConfig CR
 	tc, err = ug.operatorClient.OperatorV1alpha1().TektonConfigs().Get(ctx, v1alpha1.ConfigResourceName, metav1.GetOptions{})
 	assert.NoError(t, err)
-	assert.Equal(t, operatorVersion, tc.Status.GetAppliedUpgradeVersion())
+	assert.Equal(t, operatorVersion, tc.Status.GetAppliedPreUpgradeVersion())
+
+	// post upgrade tests
+	// ---
+	isPreUpgrade = false // post upgrade
+
+	// tektonConfig CR present, should return no error
+	err = ug.updateAppliedUpgradeVersion(ctx, isPreUpgrade)
+	assert.NoError(t, err)
+
+	// verify the post upgrade version
+	tc, err = ug.operatorClient.OperatorV1alpha1().TektonConfigs().Get(ctx, v1alpha1.ConfigResourceName, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, operatorVersion, tc.Status.GetAppliedPostUpgradeVersion())
```

Comment:
Can we add an assertion when an error occurs ?

Reply:
@PuneetPunamiya  there is an error check in line 140.
Are you pointing different one?